design/Commit/How a commit is done in FDB.md

```diff
@@ -0,0 +1,214 @@
+
+### Preresolution Section
+
+* The commit proxy sends a request for commit version, with a request number.
+
+  - The request number is a monotonically increasing number per commit proxy.
+  - This ensures for each proxy, the master will process the requests in order.
+
+* The master server waits until the request number is current.
+
+  When the current request number is larger than the incoming request number
+
+  * If a commit version is already assigned to the incoming request number, return the commit version and the previous commit version. (i.e. `prevVersion`)
+
+  * Otherwise return `Never`
+
+  * Increase current commit version, return it back to the commit proxy.
+
+    * Only one process serves as master. Thus the commit version is unique for each cluster.
+
+    * The monotonically increasing commit version will ensure that each transaction is processed in a strict serial order.
+
+### Resolution section
+
+* The commit proxy sends the transaction to the resolver.
+* Resolver waits until its version reaches `prevVersion`
+  * Ensures all transactions having version smaller than this transaction are resolved.
+  * Detects conflicts for the given transaction:
+    * If there is no conflict, return `TransactionCommitted` as the status
+    * Any conflict, return `TransactionConflict` status
+    * If the read snapshot is not in MVCC, return `TransactionTooOld` status
+
+### Post Resolution section
+
+* The proxy waits until the local batch number is current
+
+### TLog section
+* The proxy updates the metadata keys and attaches corresponding storage servers' tags to all mutations.
+* The proxy then waits until the commit version is current, i.e. the proxy's committed version is catching up with the commit version of the batch and these two versions are within the MVCC window.
+* The proxy pushs the commit data to TLog
+* TLog waits the commit version to current, then persist the commit.
+* Wait until *all* TLogs returns the transaction result.
+
+### Reply section
+
+* The proxy updates the master with the committed version for next GRV request at the master.
+* Reply the result to the client, base on the result from the resolver.
+
+## Tracking the process using `g_traceBatch`
+
+`g_traceBatch` can be used for querying the transactions and commits. A typical query string for Splunk is:
+
+```
+LogGroup=loggroup Type=type Location=location
+```
+
+The format of `location` is, in general, `<source_file_name>.<function/actor name>.<log information>`, e.g.
+
+```
+NativeAPI.getConsistentReadVersion.Before
+```
+
+means the `location` is at `NativeAPI.actor.cpp`, `ACTOR` `getConsistentReadVersion`, `Before` requesting the read version from GRV Proxy.
+
+Some of example queries are:
+
+```
+LogGroup=loggroup Type=TransactionDebug Location=NativeAPI*
+```
+
+```
+LogGroup=loggroup Type=CommitDebug Location=storageserver*
```

Comment:
I don't think there are any `CommitDebug` messages on the storage server.

Reply:
Yeah... Should correct this.